### PR TITLE
Rework cloud manager instance discovery

### PR DESCRIPTION
**Pull request by enaydanov**
- Use correct IP address for SSH connection
- Don't use support user credentials (the SSH key will be provided by siren-tests)
- Small refactoring of `find_and_append_cloud_manager_instance_to_collecting_nodes()`

Need to backport https://github.com/scylladb/scylla-cluster-tests/pull/6381/commits/e1e85b4508b227a6076920152b8d066ce382d4cb to SCT release branches to stop sync support key.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: defaults/test_default.yaml [hunk 1]
```diff
@@ -54,7 +54,6 @@ post_behavior_loader_nodes: "destroy"
 post_behavior_monitor_nodes: "keep-on-failure"
 post_behavior_k8s_cluster: "keep-on-failure"
 
-cloud_credentials_path: '~/.ssh/support'
```
Comment (fruch) [inline defaults/test_default.yaml:57]: If this is removed, let's remove it completely from sct_config

File: get-qa-ssh-keys.sh [hunk 2]
```diff
@@ -9,7 +9,6 @@ umask 077
 aws s3 sync s3://${SSH_KEYS_BUCKET} ${SSH_KEYS_DIR} --exact-timestamps --exclude '*' \
   --include scylla-qa-ec2 \
   --include scylla-test \
-  --include support \
```
Comment (fruch) [inline get-qa-ssh-keys.sh:12]: This support key isn't gonna be used at all ?

Let's check if we can remove it completely (and stop downloading it also on SCT release branches)

File: defaults/test_default.yaml (hunk 1 above)
Comment (fruch) [reply defaults/test_default.yaml:57]: If not, let's leave an empty default.

Comment (enaydanov) [reply defaults/test_default.yaml:57]: OK, I'll set it to empty string.

File: get-qa-ssh-keys.sh (hunk 2 above)
Comment (enaydanov) [reply get-qa-ssh-keys.sh:12]: I think we need to keep it in keystore, but stop to download in release branches.

Comment (fruch) [reply get-qa-ssh-keys.sh:12]: So let's put this in a separate commit so we can backport

Comment (enaydanov) [reply get-qa-ssh-keys.sh:12]: It's already done, btw :)

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**